src/content/blog/2025-10-03-whats-coming-in-eslint-10.0.0.md

```diff
@@ -0,0 +1,149 @@
+* Node.js v20 is in Maintenance LTS until July 2025
+* Node.js v22 became Active LTS in July 2025
+
+The updated Node.js requirements will enable ESLint to leverage newer JavaScript features, including native support for `require(esm)` (available since Node.js v20.19.0), and improved performance characteristics of modern Node.js versions.
+
+### Removal of eslintrc functionality
+
+In ESLint v9.0.0, we deprecated the old eslintrc config system while leaving the functionality available for backwards compatibility. In v10.0.0, we are [removing the eslintrc config system](https://eslint.org/blog/2023/10/flat-config-rollout-plans/#eslintrc-removed-in-eslint-v10.0.0) completely. Specifically, this means:
+
+1. The `ESLINT_USE_FLAT_CONFIG` environment variable is no longer honored.
+1. The CLI no longer supports eslintrc-specific arguments (`--no-eslintrc`, `--env`, `--resolve-plugins-relative-to`, `--rulesdir`, `--ignore-path`).
+1. `.eslintrc.*` and `.eslintignore` files will no longer be honored.
+1. The `ESLint` class' `configType` argument can only be `"flat"` and will throw an error if `"eslintrc"` is passed.
+1. The `shouldUseFlatConfig()` method will always return `true`.
+1. The `loadESLint()` function now always returns an instance of `ESLint`.
+1. The `Linter:verify()` method `configType` argument can only be `"flat"`.
+1. The following `Linter` eslintrc-specific methods are removed:
+    * `defineParser()`
+    * `defineRule()`
+    * `defineRules()`
+    * `getRules()`
+1. The following changes to the `/use-at-your-own-risk` entrypoint:
+    * `LegacyESLint` is removed
+    * `FileEnumerator` is removed
+    * `shouldUseFlatConfig()` method will always return `true`
+
+### Updating `eslint:recommended` configuration
+
+The `eslint:recommended` configuration will be [updated for v10.0.0](https://github.com/eslint/eslint/issues/19966) to include new rules that help catch common programming errors and improve code quality.
+
+### Removing support for jiti < 2.2.0
+
+ESLint v10.0.0 will [drop support for jiti versions prior to 2.2.0](https://github.com/eslint/eslint/issues/19765) when loading TypeScript configuration files due  known issues that can cause compatibility problems when configurations load certain plugins (as documented in [issue #19413](https://github.com/eslint/eslint/issues/19413))
+
+### Updates to `Program` AST node range coverage
+
+ESLint v10.0.0 will include a significant change to how the `Program` AST node's range is calculated, [updating it to span the entire source text](https://github.com/eslint/js/issues/648).
+
+Currently, the `Program` node's range excludes leading and trailing comments/whitespace, which creates some unintuitive scenarios:
+
+```javascript
+// Leading comment
+const x = 1;
+// Trailing comment
+```
+
+* **Current:** `Program` range covers only `const x = 1;` (excludes comments)
+* **New:** `Program` range covers the entire file from start to finish
+
+### Removing deprecated rule context methods
```

Comment:
I think we should also mention removal of the `context.parserPath` and `context.parserOptions` properties. It's a subtask of https://github.com/eslint/eslint/issues/13481.